Pass recipient AddressMetadata to send-event

ezdac opened the pull request:
Resolves: #6894

fredo reviewed (approved):
In general looks good to me. two notes, which we already discussed: 

- TransferTargetState could live without the initiator address metadata if it were stored in the `from_transfer` attribute.

- `events_for_secretreveal` is the only function where the state argument was changed instead of adding a `recipient_metadata`. Although, as I said privately, I like this solution better, the inconsistent handling is a bit weird. But I think changing this is just a matter of taste.

ezdac commented:
Ok, I changed the states and changed back the call signature for `events_for_secretreveal`.
The `LockedTransferSignedState` now stores `route_states` instead of pure address based paths,
but it still has the `routes` legacy-property that extract the address based paths from the route-states.
This is accessed quite heavily, but this can be refactored later, if so desired.

fredo commented:
Overall looks good to me
